### PR TITLE
Fix wrong precision in calculation of gas fee

### LINKED ISSUE #799: wrong precision in calculation of gas fee during call AnteHandler
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]
Code: 177574aafa34683ccd1823cdb8ca5642f271e48d

__Steps to reproduce:__

https://github.com/cosmos/ethermint/blob/177574aafa34683ccd1823cdb8ca5642f271e48d/app/ante/eth.go#L102
call `NewDecCoin ` will increment precision to 18, it means if you call it like:
```go
fee := NewDecCoin("eth",1)
```
The `fee` will become to 1000000000000000000, that's a wrong precision calculation

1. [First Step]
2. [Second Step]
3. [and so on...]

__Expected behavior:__ [What you expected to happen]

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

### PR DESCRIPTION
**Pull request by KamiD**
Closes: #799 

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by xiangjianmeng — APPROVED** [review]

**Comment by araskachoi** [comment]
@KamiD 
This fix does not fix #799. The proposed fix here will make the multiplier into 18*10^18, which will break the default precision calculation. could you revisit this and see if it is actually an issue?

**Comment by KamiD** [comment]
> @KamiD
> This fix does not fix #799. The proposed fix here will make the multiplier into 18*10^18, which will break the default precision calculation. could you revisit this and see if it is actually an issue?

hi, I have test this case in ut and debug in step by step, that is actually an issue here.
for math example:
* set gas price to 1 GWEI, set gas limit to 10000
* in AnteHandler, the code `fee = price * GasLimit` should be `10000000000000`, that's a correct value
* if I use old code, the result of fee will be 10000000000000 ^ 18 cause `NewDecCoin` will call func `ToDec()` and it will change the value to n ^ 18
* if I use the new code, the result of fee will be 10000000000000

in case of using function `NewDecCoinFromDec `, if I set the param `prec` to 0, the result will be n ^ 18, if I set `prec` to 18, the result will be n ^ 0, after debugging and running tests case on okexchain, this pr will fix the precision error.

BTW: the function `NewDecCoinFromDec` was a little anti-humanity

**Comment by araskachoi** [comment]
tested and works as intended! 
@KamiD can we write a quick precision test to make sure that the fee has the correct decimal precision? looks ready to approve/merge after that. thanks!

File: app/ante/eth.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (emfd EthMempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 	evmDenom := emfd.evmKeeper.GetParams(ctx).EvmDenom
 
 	// fee = gas price * gas limit
-	fee := sdk.NewDecCoin(evmDenom, sdk.NewIntFromBigInt(msgEthTx.Fee()))
+	fee := sdk.NewDecCoinFromDec(evmDenom, sdk.NewDecFromBigIntWithPrec(msgEthTx.Fee(), sdk.Precision))
```
Comment (fedekunze) [inline app/ante/eth.go:102]: although this is correct, this assumes the precision is always 18, after stargate is migrated we would need to query the denom metadata to get the decimals

**Review by fedekunze — APPROVED** [review]
ACK. Can you add a changelog entry?

**Comment by araskachoi** [comment]
@KamiD can you update the changelog file please? ready to merge afterwards. apologies on the delay

**Comment by KamiD** [comment]
> @KamiD can you update the changelog file please? ready to merge afterwards. apologies on the delay

hey bro, its done

File: app/ante/eth.go (hunk 1 above)
Comment (KamiD) [reply app/ante/eth.go:102]: agree

**Pull request merged**